src/main/java/uk/gov/register/util/HashValue.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.register.util;
+
+import uk.gov.register.core.HashingAlgorithm;
+
+public class HashValue {
+    private final String rawHash;
+    private final String encodedHash;
+
+    public HashValue(HashingAlgorithm hashingAlgorithm, String rawHash) {
+        this.rawHash = rawHash;
+        this.encodedHash = hashingAlgorithm.toString() + ":" + rawHash;
+    }
+
+    public String encode() {
+        return encodedHash;
+    }
+
+    public String decode() {
+        return rawHash;
+    }
```

Comment:
The purpose of the decode() method is that it takes an encoded string as a parameter and returns a `HashValue` (as per the example in Trello). i.e. it takes "sha-256:xxxxxx" and returns a HashValue with rawHash of xxxxxx.